### PR TITLE
[FLINK-17836][hive][doc] Add document for Hive dim join

### PR DESCRIPTION
**Pull request by lirui-apache**
## What is the purpose of the change

Add doc for Hive table lookup join.

## Brief change log

Add doc

## Verifying this change

NA

## Does this pull request potentially affect one of the following parts:

NA

## Documentation

NA

### CODE REVIEW

**Comment by lirui-apache** [comment]
Screen Shot 2020-06-11 at 8 51 25 PM

File: docs/dev/table/hive/hive_streaming.md [hunk 1]
```diff
@@ -163,4 +163,33 @@ SELECT * FROM hive_table /*+ OPTIONS('streaming-source.enable'='true', 'streamin
 
 ## Hive Table As Temporal Tables
 
-TODO
+Starting from Flink 1.11.0, you can use a Hive table as temporal table and join streaming data with it. Please follow
```
Comment (JingsongLi) [inline docs/dev/table/hive/hive_streaming.md:166]: I think we don't need say "starting from", users can look docs.

File: docs/dev/table/hive/hive_streaming.md [hunk 2]
```diff
@@ -163,4 +163,33 @@ SELECT * FROM hive_table /*+ OPTIONS('streaming-source.enable'='true', 'streamin
...
+the [example]({{ site.baseurl }}/dev/table/streaming/temporal_tables.html#temporal-table) to find out how to join a
+temporal table. When performing the join, the Hive table will be cached in TM memory and each record from the stream
```
Comment (JingsongLi) [inline docs/dev/table/hive/hive_streaming.md:168]: `When performing the join` can be another line

File: docs/dev/table/hive/hive_streaming.md [hunk 3]
```diff
@@ -163,4 +163,33 @@ SELECT * FROM hive_table /*+ OPTIONS('streaming-source.enable'='true', 'streamin
...
+        <td><h5>lookup.join.cache.ttl</h5></td>
+        <td style="word-wrap: break-word;">60 min</td>
+        <td>Duration</td>
+        <td>The cache TTL (e.g. 10min) for the build table in lookup join. By default the TTL is 60 minutes.</td>
+    </tr>
+  </tbody>
+</table>
+
+**Note**:
+1. You need to make sure the Hive table can fit into TM memory since the whole table will be cached.
```
Comment (JingsongLi) [inline docs/dev/table/hive/hive_streaming.md:193]: TM memory / slot > Hive table memory, you can explain this.

File: docs/dev/table/hive/hive_streaming.md [hunk 4]
```diff
@@ -163,4 +163,33 @@ SELECT * FROM hive_table /*+ OPTIONS('streaming-source.enable'='true', 'streamin
...
+2. You should set a relatively large value for `lookup.join.cache.ttl`. You'll probably have performance issue if
+your Hive table needs to be updated and reloaded too frequently.
```
Comment (JingsongLi) [inline docs/dev/table/hive/hive_streaming.md:195]: You can add a note to explain, we will read all data from all partitions, we can not read latest partition.

Comment (lirui-apache) [reply docs/dev/table/hive/hive_streaming.md:195]: I have mentioned that the whole table will be cached. And the temporal table can be either partitioned or non-partitioned. It seems to me that talking about new/old partitions here might bring more confusions.

Comment (lirui-apache) [reply docs/dev/table/hive/hive_streaming.md:198]: I have added some comments about this anyway, but w/o too many details.

**Pull request merged**